src/Language/Haskell/Liquid/GHC/Plugin.hs

```diff
@@ -231,71 +231,74 @@ typecheckHook _ (unoptimise -> modSummary) tcGblEnv = do
 liquidHaskellCheck :: PipelineData -> ModSummary -> TcGblEnv -> TcM TcGblEnv
-liquidHaskellCheck pipelineData modSummary tcGblEnv = (do
+liquidHaskellCheck pipelineData modSummary tcGblEnv = do
   cfg <- liftIO getConfig
-
-  -- The 'specQuotes' contain stuff we need from imported modules, extracted
-  -- from the annotations in their interface files.
-  let specQuotes :: [BPspec]
-      specQuotes = LH.extractSpecQuotes' tcg_mod tcg_anns tcGblEnv
-
-  -- Here, we are calling Liquid Haskell's parser, acting on the unparsed
-  -- spec comments stored in the pipeline data, supported by the specQuotes
-  -- obtained from the imported modules.
-  inputSpec :: BareSpec <-
-    (getLiquidSpec thisModule (pdSpecComments pipelineData) specQuotes)
-
-  debugLog $ " Input spec: \n" ++ show inputSpec
-  debugLog $ "Relevant ===> \n" ++ unlines (renderModule <$> S.toList (relevantModules modGuts))
-
-  logicMap :: LogicMap <- liftIO LH.makeLogicMap
-
-  -- debugLog $ "Logic map:\n" ++ show logicMap
-
-  let lhContext = LiquidHaskellContext {
-        lhGlobalCfg       = cfg
-      , lhInputSpec       = inputSpec
-      , lhModuleLogicMap  = logicMap
-      , lhModuleSummary   = modSummary
-      , lhModuleTcData    = pdTcData pipelineData
-      , lhModuleGuts      = pdUnoptimisedCore pipelineData
-      , lhRelevantModules = relevantModules modGuts
-      }
-  
-  -- liftIO $ putStrLn ("liquidHaskellCheck 6: " ++ show isIg)
-
-  liquidLib <- if isIgnore inputSpec 
-                then pure emptyLiquidLib 
-                else checkLiquidHaskellContext lhContext 
- 
-  -- --- 
-  -- -- CAN WE 'IGNORE' THE BELOW? TODO:IGNORE -- issue use `emptyLiquidLib` instead of pmrClientLib
-  -- ProcessModuleResult{..} <- processModule lhContext
-  
-  -- liftIO $ putStrLn "liquidHaskellCheck 7"
-
-  -- -- Call into the existing Liquid interface
-  -- out <- liftIO $ LH.checkTargetInfo pmrTargetInfo
-  -- liftIO $ putStrLn "liquidHaskellCheck 8"
-
-  -- -- Report the outcome of the checking
-  -- LH.reportResult errorLogger cfg [giTarget (giSrc pmrTargetInfo)] out
-  -- case o_result out of
-  --   Safe _stats -> pure ()
-  --   _           -> failM
-
-  -- liftIO $ putStrLn "liquidHaskellCheck 9"
-  -- ---
-
-  let serialisedSpec = Util.serialiseLiquidLib liquidLib thisModule
-  debugLog $ "Serialised annotation ==> " ++ (O.showSDocUnsafe . O.ppr $ serialisedSpec)
-  
-  -- liftIO $ putStrLn "liquidHaskellCheck 10"
-
-  pure $ tcGblEnv { tcg_anns = serialisedSpec : tcg_anns tcGblEnv })
-    `gcatch` (\(e :: UserError) -> LH.reportErrors Full [e] >> failM)
-    `gcatch` (\(e :: Error) -> LH.reportErrors Full [e] >> failM)
-    `gcatch` (\(es :: [Error]) -> LH.reportErrors Full es >> failM)
+  let filters = StringFilter <$> expectErrorContaining cfg
+      continue = pure tcGblEnv
+      reportErrs :: (Show e, F.PPrint e) => [TError e] -> TcM TcGblEnv
+      reportErrs = LH.filterReportErrors continue filters Full
+  (do
```

Comment:
Would it be possible to avoid changing the indentation of the code to help the review?

Perhaps it all would be more manageable if the large do block was placed in an auxiliary function.

Also, if `filters` and `reportErrs` had a `cfg` parameter they could be placed in a `where` clause, or at the top level.

Reply:
Changed.

Reply:
Thanks Matthew. The indentation changes went into a different commit though, which doesn't help observing the diff in the preexisting commit.

Also, the factoring-out of the code to a new function would deserve a standalone commit too.